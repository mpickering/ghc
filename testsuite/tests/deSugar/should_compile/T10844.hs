module T10844 where

import T10844a

-- String literals should not be inlined, the point of this test is to
-- check that the string "foo" from T10844a does not appear in the
-- simplified core of T10844.

n :: Int
n = 0
{-# NOINLINE n #-}

main = print (foo n)
