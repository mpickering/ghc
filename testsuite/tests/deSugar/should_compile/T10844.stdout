T10844a.$fFooInt2 = "foo"#
